### PR TITLE
Fix regex error

### PR DESCRIPTION
**Pull request by Guiforge**
- escape host for use in regex  8df6e73
> https://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex
- Check regex before save  bb62989

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -208,11 +208,24 @@ observe.wildcard = h => {
   }
   return h;
 };
+
+observe.escapeRegexp = str => {
+  var specials = [
```
Comment (ray-lothian) [inline common.js:213]: please use `const`

File: common.js [hunk 2]
```diff
@@ -208,11 +208,24 @@ observe.wildcard = h => {
...
+    // order matters for these
+    "-", "[", "]"
```
Comment (ray-lothian) [inline common.js:215]: to have a consistent code styling please use `'` instead of `"`

File: data/options/index.js [hunk 3]
```diff
@@ -44,6 +44,15 @@ const wildcard = h => {
   return h;
 };
 
+function isGoodRegex(regexStr) {
+  try {
+    new RegExp(regexStr, 'i')
+    return true
+  } catch (error) {
```
Comment (ray-lothian) [inline data/options/index.js:51]: code styling:
```js
try {

}
catch (error) {
}
```

File: data/options/index.js [hunk 4]
```diff
@@ -64,7 +73,11 @@ document.getElementById('add').addEventListener('submit', e => {
   e.preventDefault();
   const hostname = e.target.querySelector('input[type=text]').value;
   if (hostname) {
-    add(hostname);
+    if (hostname.startsWith('R:') && !isGoodRegex(hostname.substr(2))) {
+      // TODO notify warning
+      return ;
```
Comment (ray-lothian) [inline data/options/index.js:78]: `return;`

**Comment by Guiforge** [comment]
Adding linter configuration like eslint might be cool :D

File: data/options/index.js [hunk 5]
```diff
@@ -75,7 +76,7 @@ document.getElementById('add').addEventListener('submit', e => {
   if (hostname) {
     if (hostname.startsWith('R:') && !isGoodRegex(hostname.substr(2))) {
       // TODO notify warning
```
Comment (ray-lothian) [inline data/options/index.js:78]: I added a toast function to notify the user.

**Pull request merged**